metrix-mapping/src/main/java/com/powsybl/metrix/mapping/TimeSeriesMapper.java

```diff
@@ -247,35 +215,22 @@ private void mapToNetwork(TimeSeriesTable table, int version, int variantId, int
         if (Double.isNaN(timeSeriesValue) || Double.isInfinite(timeSeriesValue)) {
             throw new TimeSeriesMappingException("Impossible to scale down " + timeSeriesValue + " of ts " + timeSeriesName + " at time index '" + table.getTableIndex().getInstantAt(point) + "' and version " + version);
         }
+
+        LogBuilder logBuilder = new LogBuilder().type(LogType.WARNING).version(version).point(variantId).index(table.getTableIndex());
         if (Math.abs(timeSeriesValue) > 0) {
             // check equipment list is not empty
             if (mappedEquipments.isEmpty()) {
-                TimeSeriesMappingLogger.Log log = new TimeSeriesMappingLogger.EmptyFilterWarning(timeSeriesName, timeSeriesValue, table.getTableIndex(), version, point);
+                EmptyFilter emptyFilter = new EmptyFilter().timeSeriesName(timeSeriesName).timeSeriesValue(timeSeriesValue).build();
+                Log log = logBuilder.point(point).logDescription(emptyFilter).buildLog();
                 if (ignoreEmptyFilter) {
                     timeSeriesMappingLogger.addLog(log);
                 } else {
                     throw new TimeSeriesMappingException(log.getMessage());
                 }
             } else {
-                if (distributionKeySum == 0) {
-                    double distributionKey = NumberDistributionKey.ONE.getValue();
-                    for (int i = 0; i < mappedEquipments.size(); i++) {
-                        distributionKeys[i] = distributionKey;
-                        distributionKeySum += distributionKeys[i];
-                    }
-                    timeSeriesMappingLogger.addLog(new ZeroDistributionKeyInfo(timeSeriesName, timeSeriesValue, table.getTableIndex(), version, variantId,
-                            mappedEquipments.stream().map(MappedEquipment::getId).collect(Collectors.toList())));
-                }
+                distributionKeySum = logDistributionKeySumNull(mappedEquipments, timeSeriesName, distributionKeys, distributionKeySum, timeSeriesValue, logBuilder);
 
-                if (mappedEquipments.get(0).getIdentifiable() instanceof HvdcLine) {
```

Comment:
QCA : RMQ => both code have not the same behaviour. after logging the old code call "return" and not the new one